torch/nn/modules/fold.py

```diff
@@ -0,0 +1,74 @@
+import torch
+from torch.autograd import Variable
+
+from .module import Module
+from .utils import _single, _pair, _triple
+from .._functions.thnn.fold import Col2Im
+
+
+class Fold(Module):
+    """
+
+    TODO: Finish this docstring...
+
+    Reshapes the input tensor into shape :math:`[N] + output_size` by
+    de-interleaving vectors of length :math:`\prod(kernel_size)` from the batch
+    dimension and placing them into the output tensor via a sliding
+    :math:`kernel_size` block of spatial dimensions.
+
+    The output has :math:`N` rows and :math:`output_size` spatial dimensions.
+
+    | If :attr:`padding` is non-zero, then the input is implicitly
+    zero-padded on both sides by :attr:`padding` number of points
+    | :attr:`dilation` controls the spacing between the kernel points.
+    It is harder to describe, but this `link`_ has a nice visualization of what
+    dilation does.
+
+    Args:
+        kernel_size (int or tuple): the size of the sliding blocks to convert
+                                    to columns. Default: 1
+        stride (int or tuple): the stride of the sliding blocks in the input
+                               spatial dimensions. Default: 1
+        padding (int or tuple, optional): implicit zero padding to be added on
+                                          both sides of input. Default: 0
+        dilation (int or tuple, optional): a parameter that controls the
+                                           stride of elements within the
+                                           neighborhood. Default: 1
+
+    Shape:
+        - Input: :math:`(N, C, L_{in})`
+        - Output: :math:`(N * C * \prod(kernel_size), L_{out},)` where
+          :math:`L_{out} = floor((L_{in} + 2 * padding - dilation * (kernel_size - 1) - 1) / stride + 1)
+
+    Examples::
+
+        >>> # kernel_size (2, 2), dilation (1, 1), padding (0, 0), stride (1, 1)
+        >>> unfold = nn.Unfold((2, 2), (1, 1), (0, 0), (1, 1))
+        >>> input = autograd.Variable(torch.randn(1, 256, 256))
+        >>> output = unfold(input)
+
+    .. _link:
+        https://github.com/vdumoulin/conv_arithmetic/blob/master/README.md
+
+    """
+
+    def __init__(self, output_size, kernel_size, dilation, padding, stride):
+        super(Fold, self).__init__()
+        self.output_size = output_size
+        self.kernel_size = kernel_size
+        self.dilation = dilation
+        self.padding = padding
+        self.stride = stride
+
+    def forward(self, input):
+        return Col2Im.apply(input, self.output_size, self.kernel_size, self.dilation,
```

Comment:
This should call the `functional` version, which will handle all dimensions (see `upsample` in `functional.py` for an example):
```python
def forward(self, input):
    return F.fold(input, self.output_size, self.kernel_size, self.dilation, self.padding, self.stride)
```

Reply:
and done